cpp/src/arrow/csv/writer.cc

```diff
@@ -81,7 +81,15 @@ RecordBatchIterator RecordBatchSliceIterator(const RecordBatch& batch,
 
 // Counts the number of quotes in s.
 int64_t CountQuotes(arrow::util::string_view s) {
-  return static_cast<int64_t>(std::count(s.begin(), s.end(), '"'));
+  // std::count uses 64 bit counter, not necessary for StringType
```

Comment:
What about LargeString?

Reply:
An easy approach might be to check s.size(), if it can fit in 32 bits, go the 32 bit code, otherwise call std::count directly.
Per my test, there's no performance loss, probably because the branch is almost always correctly predicted.
Will try refine this code, at least it should static_assert if we use LargeString.

Reply:
Restored to `std::count`. The improvement is not obvious overall, not deserves the effort.